### DISCUSSION TITLE
Examples for controlling the window?

### DISCUSSION THREAD

**Discussion by JonahPlusPlus** [hecrj/iced#991] [Q&A] [answered]
I am trying to implement a custom title bar with minimize, restore and close buttons. I got the window to close by using the `should_close` function hooked up to the Application State, but can't find anything for minimize and restore. Are these supported? And how would I go about moving the window by dragging it? Can an example be added for this? 

Also, I am implementing a TitleBar struct that has buttons and a view function. But it is kinda messy since you got to pass a Theme to it and apply it to all the Buttons individually. Is there a way to just recursively apply it, or is this the only way? Thanks

**Comment by akhilman** [comment]
If there will be a way to get a winit's window from iced, then we will able to drag window by `window.drag_window()` ([see example](https://github.com/rust-windowing/winit/blob/master/examples/drag_window.rs)), but winit's window seems is not exposed.

**Comment by tarkah** [comment, marked as answer]
FYI, there is an enhancement coming soon in https://github.com/hecrj/iced/pull/1019.

It updates `Command` to accept an `Action`. These actions are intended to do exactly this. The following window actions are added in the PR and will trigger the appropriate `winit` method behind the scenes. It should be extremely easy to add `Minimize`, `Restore` and any other window actions that can be called in `winit`.

https://github.com/hecrj/iced/blob/7cb6e7438f7fb5d0d8be4528a31b888e2b12cd51/native/src/window/action.rs#L1-L18
